File: src/main/scala/arimitsu/sf/cql/v3/messages/Supported.scala
package arimitsu.sf.cql.v3.messages


import arimitsu.sf.cql.v3.util.Notation
import java.nio.ByteBuffer

/**
 * Created by sxend on 14/06/07.
 */
object SupportedParser extends ResponseParser[Supported] {
  override def parse(implicit body: ByteBuffer): Supported = {
    val map = Notation.getStringMultiMap
    Supported(map)
  }
}

object Supported {
  val COMPRESSION = "COMPRESSION"
  val CQL_VERSION = "CQL_VERSION"

}

case class Supported(parameters: Map[String, List[String]])

File: src/main/scala/arimitsu/sf/cql/v3/messages/Supported.java
package arimitsu.sf.cql.v3.messages;


import arimitsu.sf.cql.v3.Parser;
import arimitsu.sf.cql.v3.util.Notation;

import java.nio.ByteBuffer;
import java.util.List;
import java.util.Map;

/**
 * Created by sxend on 14/06/07.
 */
public class Supported {

    public static final String COMPRESSION = "COMPRESSION";
    public static final String CQL_VERSION = "CQL_VERSION";

    public final Map<String, List<String>> stringMultiMap;

    public Supported(Map<String, List<String>> stringMultiMap) {
        this.stringMultiMap = stringMultiMap;
    }

}

